From c7766997f34395d561791ccb63b0c3fb60d47fb6 Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@nxp.com>
Date: Fri, 19 Jan 2018 17:44:00 +0800
Subject: [PATCH 3229/5242] MLK-17428-7: ASoC: imx-ak5558: support 768KHz
 sample rate

commit  20e1d927e14892e583f091efd94042a1c906d535 from
https://source.codeaurora.org/external/imx/linux-imx.git

For 768kHz sample rate, the codec can't support 64fs mclk, only
can support 32fs mclk, so we can't fix the slot_width to 32, which
is for S32_LE, use params_physical_width(params) to instead of
hard code.

Signed-off-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/imx-ak5558.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/fsl/imx-ak5558.c b/sound/soc/fsl/imx-ak5558.c
index 59dc901..9d936e2 100644
--- a/sound/soc/fsl/imx-ak5558.c
+++ b/sound/soc/fsl/imx-ak5558.c
@@ -59,7 +59,7 @@ static int imx_aif_hw_params(struct snd_pcm_substream *substream,
 
 	ret = snd_soc_dai_set_tdm_slot(cpu_dai,
 				       BIT(channels) - 1, BIT(channels) - 1,
-				       2, 32);
+				       2, params_physical_width(params));
 	if (ret) {
 		dev_err(dev, "failed to set cpu dai tdm slot: %d\n", ret);
 		return ret;
-- 
1.7.9.5

